src/monobase/build.sh

```diff
@@ -40,10 +40,13 @@ fi
 log "Running builder..."
 uv run --python "$MONOBASE_PYTHON" python -m monobase.build "$@"
 
-# Inside K8S
-# Write done file to signal pod ready
-# Sleep keep pod alive
+# When running in K8S:
+# - Write done file to signal pod ready
+# - Write the done timestamp to node feature discovery label
+# - Sleep keep pod alive
 if [ -n "${KUBERNETES_SERVICE_HOST:-}" ]; then
     touch "$DONE_FILE"
-    sleep 86400
+    printf "done=%s\n" "$(date --iso-8601=seconds --utc)" |
+        tee /etc/kubernetes/node-feature-discovery/features.d/monobase
```

Comment:
IIUC, this file is persisted even if monobase pod restarts? So existence of the file == monobase has "done" at least once i.e. at least 1 generation is available?

Reply:
That was my intent, yes. I can see how having a label per generation might be better. WDYT?

Reply:
That was my intent, yes. I can see how having a label per generation might be better. WDYT?

Reply:
How big is a given generation identifier, we have ~256KiB of space for a single annotation, we could add all of the metadata into that single annotation and then use a scheduling extension to [`preScore`](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/#pre-score) nodes that have been filtered to break apart that annotation into scorable data (e.g. so we can raise the score of the nodes with the proper generations). 

Additionally we can lean on a [`preBind` ](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/#pre-bind) scheduler extension (that calls an api on the target node) to mutate the pod spec back to non-weightless if the target generation does not exist on the node. This will mitigate the problems with needing 3 failed setupruns to know if a generation exists on the node.

I looked into the previously discussed `mutating webhook` and that cannot address these changes because the mutating webhook acts before scheduling decisions.

Reply:
> That was my intent, yes. I can see how having a label per generation might be better. WDYT?
Not necessarily. Generations are not versions, merely ways to simulate atomicity on FS. Latest should always work and not break from previous ones.
I don't wanna end up in a world where nodes have different subset of gens and models have to pin different gens like everything else we have today...

> if the annotation and label are cleared if monobase daemonset restarts
Yeah right `build.py` should short circuit if all requested generations already exists, and re-create this label.
However, "adding" a new generation might take ~20 min, during which `build.sh` is not "done" but a previous generation exists and is in a usable state. We might want to write this label when **at least one** generation is ready instead of **all** are ready.